rules-base/api/src/main/java/org/windup/rules/apps/condition/FileContent.java

```diff
@@ -0,0 +1,229 @@
+package org.windup.rules.apps.condition;
+
+import java.io.FileReader;
+import java.io.Reader;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
+
+import org.jboss.windup.config.GraphRewrite;
+import org.jboss.windup.config.Variables;
+import org.jboss.windup.config.condition.EvaluationStrategy;
+import org.jboss.windup.config.condition.NoopEvaluationStrategy;
+import org.jboss.windup.config.parameters.FrameContext;
+import org.jboss.windup.config.parameters.FrameCreationContext;
+import org.jboss.windup.config.parameters.ParameterizedGraphCondition;
+import org.jboss.windup.graph.model.WindupVertexFrame;
+import org.jboss.windup.graph.model.resource.FileModel;
+import org.jboss.windup.graph.service.FileModelService;
+import org.jboss.windup.graph.service.GraphService;
+import org.jboss.windup.util.Logging;
+import org.ocpsoft.rewrite.context.EvaluationContext;
+import org.ocpsoft.rewrite.param.DefaultParameterStore;
+import org.ocpsoft.rewrite.param.ParameterStore;
+import org.ocpsoft.rewrite.param.ParameterizedPatternResult;
+import org.ocpsoft.rewrite.param.RegexParameterizedPatternParser;
+import org.ocpsoft.rewrite.util.Maps;
+import org.windup.rules.apps.condition.regex.StreamRegexMatchListener;
+import org.windup.rules.apps.condition.regex.StreamRegexMatchedEvent;
+import org.windup.rules.apps.condition.regex.StreamRegexMatcher;
+import org.windup.rules.apps.model.FileLocationModel;
+
+import com.googlecode.streamflyer.core.Modifier;
+import com.googlecode.streamflyer.core.ModifyingReader;
+
+/**
+ * Matches on file contents based upon parameterization.
+ * 
+ * Example:
+ * 
+ * <pre>
+ *   {@link FileContent}.matches("file {number}").inFilesNamed("{filename}")
```

Comment:
Good point... it was always a number earlier, but that has changed in order to test some additional scenarios. Will fix.